astropy/nddata/tests/test_utils.py

```diff
@@ -292,6 +292,14 @@ def test_extract_array_return_pos():
         assert new_pos == (expected, )
 
 
+def test_extract_array_nan_fillvalue():
+    if minversion(np, '1.20'):
+        with pytest.raises(ValueError) as err:
```

Comment:
could be `with pytest.raises(ValueError, match='fill_value cannot be set to np.nan if'):`

Reply:
Changed, thanks!